test/nexus_test.go

```diff
@@ -350,6 +353,20 @@ func TestNexusSyncOperation(t *testing.T) {
 			tc.requireFailureCounter(t, service.Name, syncOp.Name(), "handler_error_INTERNAL")
 		}, time.Second*3, time.Millisecond*100)
 	})
+
+	t.Run("timeout", func(t *testing.T) {
+		_, err := nexus.ExecuteOperation(ctx, nc, syncOp, "timeout", nexus.ExecuteOperationOptions{
+			// Force shorter timeout to speed up the test and get a response back.
+			Header: nexus.Header{nexus.HeaderRequestTimeout: "300ms"},
+		})
+		var unexpectedResponseErr *nexus.UnexpectedResponseError
+		require.ErrorAs(t, err, &unexpectedResponseErr)
+		require.Equal(t, nexus.StatusDownstreamTimeout, unexpectedResponseErr.Response.StatusCode)
+
+		require.EventuallyWithT(t, func(t *assert.CollectT) {
+			tc.requireFailureCounter(t, service.Name, syncOp.Name(), "timeout")
```

Comment:
Why only assert the failure counter? shouldn't `NexusTaskScheduleToStartLatency` be published?

Reply:
I figured it's covered in the other tests.

Reply:
hm I think this case is different because we don't report E2E latency on timeout

Reply:
I know, I meant the other two metrics. In any case, I've added the tests.